docs/attacks/linear_regression.rst

```diff
@@ -0,0 +1,116 @@
+
+For Arbiter PUFs, the reliability for any given challenge :math:`c` has a close relationship with the difference in
+delay for the top and bottom line. When modeling the Arbiter PUF response as
+
+.. math::
+    r = \text{sgn}\left[ D_\text{noise} + \langle w, x \rangle \right],
+
+where :math:`x` is the feature vector corresponding to the challenge :math:`c` and :math:`w \in \mathbb{R}^n` are the
+weights describing the Arbiter PUF, and :math:`D_\text{noise}` is chosen from a Gaussian distribution with zero mean
+and variance :math:`\sigma_\text{noise}^2` to model the noise, then we can conclude that
+
+.. math::
+    \text{E}[r(x)] = \text{erf}\left( \frac{\langle w, x \rangle}{\sqrt{2}\sigma_\text{noise}} \right).
+
+Hence, the delay difference :math:`\langle w, x \rangle` can be approximated based on an approximation of
+:math:`\text{E[r(x)]}`, which can be easily obtained by an attacker. It gives
+
+.. math::
+    \langle w, x \rangle = \sqrt{2}\sigma_\text{noise} \cdot \text{erf}^{-1} \text{E}[r(x)].
+
+This approximation works well even when :math:`\text{E}[r(x)]` is approximated based on only on few responses, e.g. 3
+(see below).
+
+To demonstrate the attack, we initialize an Arbiter PUF simulation with noisiness chosen such that the reliability
+will be about 91% *on average*:
+
+>>> import pypuf.simulation, pypuf.io, pypuf.attack, pypuf.metrics
+>>> puf = pypuf.simulation.ArbiterPUF(n=64, noisiness=.25, seed=3)
+>>> pypuf.metrics.reliability(puf, seed=3).mean()
+0.908...
+
+We then create a CRP set using the *average* value of responses to 500 challenges, based on 5 measurements:
+
+>>> challenges = pypuf.io.random_inputs(n=puf.challenge_length, N=500, seed=2)
+>>> responses_mean = puf.r_eval(5, challenges).mean(axis=-1)
+>>> crps = pypuf.io.ChallengeResponseSet(challenges, responses_mean)
+
+Based on these approximated values ``responses_mean`` of the linear function :math:`\langle w, x \rangle`, we use
+linear regression to find a linear mapping with small error to fit the data. Note that we use the ``transform_atf``
+function to compute the feature vector :math:`x` from the challenges :math:`c`, as the mapping is linear in :math:`x`
+(but not in :math:`c`).
+
+>>> attack = pypuf.attack.LeastSquaresRegression(crps, feature_map=lambda cs: pypuf.simulation.ArbiterPUF.transform_atf(cs, k=1)[:, 0, :])
+>>> model = attack.fit()
+
+The linear map ``model`` will predict the delay difference of a given challenge. To obtain the predicted PUF response,
+this prediction needs to be thresholded to either -1 or 1:
+
+>>> model.postprocessing = model.postprocessing_threshold
+
+To measure the resulting model accuracy, we use :meth:`pypuf.metrics.similarity`:
+
+>>> pypuf.metrics.similarity(puf, model, seed=4)
+array([0.902])
+
+
+Modeling Attack on Integrated Optical PUFs [RHUWDFJ13]_
+-------------------------------------------------------
+
+The electromagnetic field of an integrated optical PUF token can be understood as a linear map
```

Comment:
While the math makes sense this way, I find the wording here, calling the EM-field amplitudes a "map", a bit confusing. Below, it is pointed out that the linear map is a model of the PUF. Thus, it is confusing to call the EM-amplitudes (which is the result of applying the map to a challenge) by the same name.